### PR TITLE
GH-1691C: As a smith I want to improve "add missing imports" functionality in code completion and quick fixes

### PR DESCRIPTION
**Pull request by mor-n4**
See #1691.

Includes:
- bug fix for "add missing import" quick fix (now works with CMD+`.` in VS Code),
- change organize imports to add missing imports,
- major clean up of the old import-related code.

### CODE REVIEW

File: tests/org.eclipse.n4js.lang.tests/src/org/eclipse/n4js/ide/server/codeActions/util/ChangeProviderTest.xtend [hunk 1]
```diff
@@ -21,6 +21,30 @@ import org.junit.Test
  */
 class ChangeProviderTest extends AbstractN4JSTest {
```
Comment (mmews-n4) [inline tests/org.eclipse.n4js.lang.tests/src/org/eclipse/n4js/ide/server/codeActions/util/ChangeProviderTest.xtend:22]: Good tests!

File: tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/server/AbstractCodeActionTest.java [hunk 2]
```diff
@@ -71,9 +75,48 @@ protected void performTest(Project project, String moduleName, TestCodeActionCon
...
-			String resultStr = Strings.toString(getStringLSP4J()::toString3, result);
+			String resultStr = result.stream()
+					.map(cmdOrAction -> getStringLSP4J().toString3(cmdOrAction))
+					.collect(Collectors.joining("\n-----\n"));
 			assertEquals(tcac.getExpectedCodeActions().trim(), resultStr.trim());
 		}
 	}
 
+	/** Like {@link TestCodeActionConfiguration}, but non-empty ranges can be defined. */
+	public static class TestCodeActionConfiguration2 extends TestCodeActionConfiguration {
```
Comment (mmews-n4) [inline tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/server/AbstractCodeActionTest.java:86]: Why `...2`? Maybe there would be a more telling name (sometimes we use `N4JS...`).

File: releng/org.eclipse.n4js.targetplatform/N4JS.setup [hunk 3]
```diff
@@ -236,7 +236,7 @@
         name="org.eclipse.xpect.sdk.feature.group"
         versionRange="[0.2.0,0.2.1)"/>
     <repository
-        url="https://ci.eclipse.org/xpect/job/Xpect-Integration-Release/20/artifact/org.eclipse.xpect.releng/p2-repository/target/repository/"/>
+        url="https://ci.eclipse.org/xpect/job/Xpect-Integration-Release/22/artifact/org.eclipse.xpect.releng/p2-repository/target/repository/"/>
```
Comment (mmews-n4) [inline releng/org.eclipse.n4js.targetplatform/N4JS.setup:239]: Thanks for updating this along with this PR!

**Comment by mor-n4** [comment]
Builds green-ish, merging ...

**Pull request merged**